### PR TITLE
docs: #717 tiered storage delete behavior

### LINKED ISSUE #717: Document new tiered storage data deletion behavior in 22.3
### Describe the Issue

Redpanda didn't used to delete data in S3 when topics were deleted: this is fixed in Redpanda 22.3, along with a new topic property for toggling this deletion behavior, and a safe upgrade path where legacy topics have S3 deletion disabled by default.

### Updates to existing documentation

_Aside: the below details make this sound more complicated than it is: the basic "normal" user behavior here is that when they delete a topic, the topic is deleted: simple.  The complexity all comes from the special handling of legacy topics and the niche option to skip deleting data in S3 when a topic is deleted, which is only needed under special circumstances (e.g. when intervening in an unhealthy cluster and planning on using some external tools to copy+recover from the data in S3, or when using read replicas and deleting the topic from an original cluster while keeping it available for read replicas)._

In previous versions of Redpanda, we wrote data into S3 when using tiered storage, but never deleted it.  When a topic was deleted, the data in S3 related to that topic remained there.  There was no automatic way to remove that data.  Removing it by hand is possible but inadvisable, given the risk of a mistake damaging other topics' data.

In PR https://github.com/redpanda-data/redpanda/pull/6683 there is a new topic configuration property:
- `redpanda.remote.delete` may be set to "true" or "false".  This property appears in output of commands like `rpk topic describe` and may be set with the same `rpk` commands that modify other topic properties (e.g. how we would set `redpanda.remote.write` on a topic.
- `redpanda.remote.delete` is set to `false` on topics that were created with earlier versions of Redpanda.  This preserves the legacy behavior.  This applies only to topics created using old versions of redpanda, not to topics created on clusters after they are upgraded.
- For topics created using Redpanda 22.3 or later, the default value of `redpanda.remote.delete` is `true`.

When a topic is deleted, the topic's data in S3 is also deleted if all of the following are true:
- `redpanda.remote.delete` is true
- Full tiered storage is enabled for the topic (i.e. both `redpanda.remote.read` and `redpanda.remote.write` are true, or their respective cluster-wide defaults are both true).
- The topic is not a read replica topic.

The default behavior for new topics on clusters using tiered storage is intuitive: the `rpk topic delete` command deletes the topic and its data, whether that data is on local storage or in S3.

Legacy topics may be switched to the new behavior by setting `redpanda.remote.delete` to true on each such topic.  After doing this, the topic's data in S3 will be deleted when the topic is deleted in future.

If you wish to delete a topic without removing its data in S3, set `redpanda.remote.delete` to false before deleting the topic.

### PR DESCRIPTION
**Pull request by micheleRP**
Resolves #717 

This add info about redpanda.remote.delete

Preview:
- https://deploy-preview-1047--redpanda-documentation.netlify.app/docs/manage/tiered-storage/#enable-tiered-storage-for-a-topic
- https://deploy-preview-1047--redpanda-documentation.netlify.app/docs/manage/tiered-storage/#remote-write

### CODE REVIEW

**Comment by micheleRP** [comment]
@jcsp, @Feediver1: If this looks OK, could you please approve it?

File: docs/manage/tiered-storage.mdx [hunk 1]
```diff
@@ -127,13 +127,14 @@ When you enable Tiered Storage on a cluster, it becomes enabled for already-exis
...
-Tiered Storage also uses the following topic configuration flags:
+Tiered Storage uses the following topic-level properties:
 
-| Flag      | Description |
+| Property      | Description |
 | ----------- | ----------- |
 | `redpanda.remote.write` | Uploads data from Redpanda to cloud storage. Overrides the cluster-level `cloud_storage_enable_remote_write` configuration for the topic. |
 | `redpanda.remote.read` | Fetches data from cloud storage to Redpanda. Overrides the cluster-level `cloud_storage_enable_remote_read` configuration for the topic. |
 | `redpanda.remote.recovery` | Recovers or reproduces a topic from cloud storage. Use this flag during topic creation. It does not apply to existing topics.  |
+| `redpanda.remote.delete` | Deletes a topic and all topic data, including data in remote storage. To delete a topic but retain data in remote storage, set to `false` before deleting the topic. This property requires that both remote read and remote write are enabled, and the topic must not be a Remote Read Replica topic.<br/><br/>Default is `true` for new topics, including new topics on upgraded clusters. Default is `false` for topics created prior to release 22.3. |
```
Comment (jcsp) [inline docs/manage/tiered-storage.mdx:137]: The tense is a bit odd: when I read "Deletes a topic and all topic data" it sounds like maybe _setting_ this property causes the data to be deleted immediately.

I would tend to explain it as "When set to true, deleting a topic will also delete its objects in remote storage.  When set to false, objects in remote storage will be left behind when the topic is deleted"

Comment (micheleRP) [reply docs/manage/tiered-storage.mdx]: Thank you, @jcsp! Here's the revised text: https://deploy-preview-1047--redpanda-documentation.netlify.app/docs/manage/tiered-storage/#enable-tiered-storage-for-a-topic

File: docs/manage/tiered-storage.mdx [hunk 2]
```diff
@@ -240,7 +241,9 @@ rpk topic alter-config <topic_name> --set redpanda.remote.write=true
 
 If remote write is enabled, log segments are not deleted until they’re uploaded to remote storage. Because of this, the log segments may exceed the configured retention period until they’re uploaded, so the topic might require more disk space. This prevents data loss if segments cannot be uploaded fast enough or if the retention period is very short.
 
-Starting in release 22.3, when you delete a topic in Redpanda, the data is also deleted in cloud storage. Topics created prior to 22.3 can alter their configuration to this behavior on a topic level after upgrading.
+:::note
+Starting in release 22.3, when you delete a topic in Redpanda, the data is also deleted in cloud storage. See [Enable Tiered Storage for a topic](../../manage/tiered-storage/#enable-tiered-storage-for-a-topic)
```
Comment (Feediver1) [inline docs/manage/tiered-storage.mdx:245]:
```suggestion
Starting in version 22.3, when you delete a topic in Redpanda, the data is also deleted in cloud storage. See [Enable Tiered Storage for a topic](../../manage/tiered-storage/#enable-tiered-storage-for-a-topic)

Want to be consistent with our doc UI, which shows "versions" not "releases"
```

**Review by Feediver1 — APPROVED** [review]

File: docs/manage/tiered-storage.mdx (hunk 2 above)
Comment (Feediver1) [reply docs/manage/tiered-storage.mdx:245]: Want to be consistent with our doc UI, which shows "versions" not "releases"

**Comment by micheleRP** [comment]
@jcsp: If you don't have any more feedback for this, could you pls approve it?

**Review by jcsp — APPROVED** [review]

**Pull request merged**